rg3d-core/src/pool.rs

```diff
@@ -1052,16 +1064,14 @@ impl<'a, T> Iterator for PoolPairIterator<'a, T> {
 
     fn next(&mut self) -> Option<Self::Item> {
         loop {
-            match self.pool.records.get(self.current) {
-                Some(record) => {
-                    if let Some(payload) = &record.payload {
-                        let handle = self.pool.handle_from_index(self.current);
-                        self.current += 1;
-                        return Some((handle, payload));
-                    }
+            if self.pool.records.get(self.current).is_some() {
+                if let Some(pair) = self.pool.at_pair(self.current) {
```

Comment:
I think iterator has become less efficient with this, because of multiple checks of same condition:

1) `.get(self.current).is_some()` checks if index is valid
2) Current implementation of `at_pair` checks index two times

However, existing implementation of iterator does checks two times - it does same checks in `handle_from_index` at line 1058. At this point we've passed all checks already and maybe it is worth to just set `handle = Handle { generation: record.generation, index: self.current as u32 };`